pkg/catalog/catalog.go

```diff
@@ -9,55 +9,64 @@ import (
 )
 
 // List lists all the available component types
-func List(client *occlient.Client) ([]string, error) {
+// returned map has key: imagestream and template and value is slice of respective components
+func List(client *occlient.Client) (map[string][]string, error) {
 	var catalogList []string
 	imageStreams, err := getDefaultBuilderImages(client)
 	if err != nil {
 		return nil, errors.Wrap(err, "unable to get image streams")
 	}
 	catalogList = append(catalogList, imageStreams...)
-
+	finalList := make(map[string][]string)
+	finalList["imagestream"] = catalogList
 	// TODO: uncomment when component create supports template creation
-	//clusterServiceClasses, err := client.GetClusterServiceClassExternalNames()
-	//if err != nil {
-	//	return nil, errors.Wrap(err, "unable to get cluster service classes")
-	//}
-	//catalogList = append(catalogList, clusterServiceClasses...)
+	clusterServiceClasses, err := client.GetClusterServiceClassExternalNames()
+	if err != nil {
+		log.Debug("Unable to get cluster serviceClassExternalName")
```

Comment:
I believe you forgot to add an error out here.. At the moment, this prints out the debug and then continues.

Reply:
I purposefully did that, because even if service catalog is not there, `odo catalog list` should work.
if there is service catalog but it's not listing the services, user can check debug logs